packages/database/src/core/SyncPoint.ts

```diff
@@ -237,71 +222,57 @@ export class SyncPoint {
     return { removed: removed, events: cancelEvents };
   }
 
-  /**
-   * @return {!Array.<!View>}
-   */
   getQueryViews(): View[] {
-    const values = Object.keys(this.views_).map(key => this.views_[key]);
-    return values.filter(function(view) {
-      return !view
-        .getQuery()
-        .getQueryParams()
-        .loadsAllData();
-    });
+    return Array.from(this.views.values()).filter(
+      view =>
+        !view
+          .getQuery()
+          .getQueryParams()
+          .loadsAllData()
+    );
   }
 
   /**
-   *
-   * @param {!Path} path The path to the desired complete snapshot
-   * @return {?Node} A complete cache, if it exists
+   * @param path The path to the desired complete snapshot
+   * @return A complete cache, if it exists
    */
   getCompleteServerCache(path: Path): Node | null {
     let serverCache: Node | null = null;
-    forEach(this.views_, (key: string, view: View) => {
+    for (const view of this.views.values()) {
       serverCache = serverCache || view.getCompleteServerCache(path);
-    });
+    }
     return serverCache;
   }
 
-  /**
-   * @param {!Query} query
-   * @return {?View}
-   */
   viewForQuery(query: Query): View | null {
     const params = query.getQueryParams();
     if (params.loadsAllData()) {
       return this.getCompleteView();
     } else {
       const queryId = query.queryIdentifier();
-      return safeGet(this.views_, queryId);
+      return this.views.get(queryId);
     }
   }
 
-  /**
-   * @param {!Query} query
-   * @return {boolean}
-   */
   viewExistsForQuery(query: Query): boolean {
     return this.viewForQuery(query) != null;
   }
 
-  /**
-   * @return {boolean}
-   */
   hasCompleteView(): boolean {
     return this.getCompleteView() != null;
   }
 
-  /**
-   * @return {?View}
-   */
   getCompleteView(): View | null {
-    const completeView = findValue(this.views_, (view: View) =>
-      view
-        .getQuery()
-        .getQueryParams()
-        .loadsAllData()
-    );
-    return completeView || null;
+    for (const view of this.views.values()) {
```

Comment:
This creates a a copy.

Reply:
`this.views` is a `Map`. This returns an iterator, not a copy.